go/certmanager/main.go

```diff
@@ -0,0 +1,288 @@
+					Attr: fuse.Attr{
+						Mode:  0664,
+						Owner: r.Owner,
+					},
+				}, fs.StableAttr{Mode: syscall.S_IFLNK})
+				r.AddChild(certType+"_"+eapkey+"_"+tlskey+".key", key, false)
+
+				ca := r.NewPersistentInode(ctx, &fs.MemSymlink{
+					Data: r.certificates[eapkey][tlskey]["ca"],
+					Attr: fuse.Attr{
+						Mode:  0664,
+						Owner: r.Owner,
+					},
+				}, fs.StableAttr{Mode: syscall.S_IFLNK})
+				r.AddChild(certType+"_"+eapkey+"_"+tlskey+".pem", ca, false)
+
+			} else {
+
+				RegularFile[eapkey][tlskey][certType]["cert"] = &MemRegularFile{
+					Data: r.certificates[eapkey][tlskey]["cert"],
+					Attr: fuse.Attr{
+						Mode:  0664,
+						Owner: r.Owner,
+					},
+				}
+
+				certfile := r.NewPersistentInode(ctx, RegularFile[eapkey][tlskey][certType]["cert"], fs.StableAttr{Ino: Inode})
+				r.AddChild(certType+"_"+eapkey+"_"+tlskey+".crt", certfile, false)
+
+				Inode++
+				RegularFile[eapkey][tlskey][certType]["key"] = &MemRegularFile{
+					Data: r.certificates[eapkey][tlskey]["key"],
+					Attr: fuse.Attr{
+						Mode:  0664,
+						Owner: r.Owner,
+					},
+				}
+				keyfile := r.NewPersistentInode(ctx, RegularFile[eapkey][tlskey][certType]["key"], fs.StableAttr{Ino: Inode})
+				r.AddChild(certType+"_"+eapkey+"_"+tlskey+".key", keyfile, false)
+				Inode++
+
+				RegularFile[eapkey][tlskey][certType]["pem"] = &MemRegularFile{
+					Data: r.certificates[eapkey][tlskey]["ca"],
+					Attr: fuse.Attr{
+						Mode:  0664,
+						Owner: r.Owner,
+					},
+				}
+				cafile := r.NewPersistentInode(ctx, RegularFile[eapkey][tlskey][certType]["pem"], fs.StableAttr{Ino: Inode})
+				r.AddChild(certType+"_"+eapkey+"_"+tlskey+".pem", cafile, false)
+				Inode++
+			}
+		}
+	}
+	r.RegularFile = RegularFile
+}
+
+// Getattr function
+func (r *CertStore) Getattr(ctx context.Context, fh fs.FileHandle, out *fuse.AttrOut) syscall.Errno {
+	out.Mode = 0755
+	return 0
+}
+
+var _ = (fs.NodeGetattrer)((*CertStore)(nil))
+var _ = (fs.NodeOnAdder)((*CertStore)(nil))
+
+var ctx = context.Background()
+
+// NotifySystemd notify systemd
+func NotifySystemd(msg string) {
+	_, err := daemon.SdNotify(false, msg)
+	if err != nil {
+		log.LoggerWContext(context.Background(), fmt.Sprintf("Error sending systemd ready notification: %s", err.Error()))
+	}
+}
+
+// RefreshPfconfig refresh the pfconfig pool
+func (r *CertStore) RefreshPfconfig(ctx context.Context) {
+	id, err := pfconfigdriver.PfconfigPool.ReadLock(ctx)
+	if err == nil {
+		defer pfconfigdriver.PfconfigPool.ReadUnlock(ctx, id)
+
+		// We launch the refresh job once, the first time a request comes in
+		// This ensures that the pool will run with a context that represents a request (log level for instance)
+		r.refreshLauncher.Do(func() {
+			ctx := ctx
+			go func(ctx context.Context) {
+				for {
+					pfconfigdriver.PfconfigPool.Refresh(ctx)
+					time.Sleep(1 * time.Second)
+				}
+			}(ctx)
+		})
+	} else {
+		panic("Unable to obtain pfconfigpool lock in certmanager")
```

Comment:
Not sure you should panic here, logging the error should be enough

Reply:
To be honest i did a copy and paste from the function in pfdns.
I will log instead of panic.